frontend/packages/dev-console/src/components/import/import-submit-utils.ts

```diff
@@ -299,6 +299,9 @@ export const createOrUpdateBuildConfig = (
   const buildConfig = mergeData(originalBuildConfig, newBuildConfig);
   if (verb === 'update') {
     buildConfig.metadata.name = originalBuildConfig.metadata.name;
+    if (!secretName) {
```

Comment:
I inspected the `import-submit-utils` file. After debugging, reached the following conclusion:

Let's take the scenario where a user has added a secret and has deployed the application. Now the user wants to remove the secret. For this, the form is updated, this time the user selects the `No Secret` option from the Secret dropdown.
In this scenario, the `originalBuildConfig` object had the `sourceSecret` object, which mentioned the name of the secret it was holding before. While the `newBuildConfig` object, did not have the `sourceSecret` object as it was removed this time. 
But on line 299, we see that both of these, are merged together. Therefore, in spite the `newBuildConfig` did not contain the `sourceSecret` object, still the final `buildConfig` contained the `sourceSecret` object as it got it from the `originalBuildConfig` .

Therefore, it was necessary to manually delete that `sourceSecret` object from the final `buiildConfig` object before passing it to the K8sUpdate function.

Reply:
@Lucifergene instead of doing this here you can do this in `mergeData` util as we handle all the other overrides there as well.

Reply:
> @Lucifergene instead of doing this here you can do this in `mergeData` util as we handle all the other overrides there as well.

@Lucifergene I see this comment is not yet resolved. Please resolve the comment and if you have any issues LMK.